pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java

```diff
@@ -0,0 +1,126 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.minion.tasks.purge;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.pinot.common.data.Segment;
+import org.apache.pinot.common.metadata.segment.SegmentZKMetadata;
+import org.apache.pinot.controller.helix.core.minion.ClusterInfoAccessor;
+import org.apache.pinot.controller.helix.core.minion.generator.PinotTaskGenerator;
+import org.apache.pinot.controller.helix.core.minion.generator.TaskGeneratorUtils;
+import org.apache.pinot.core.common.MinionConstants;
+import org.apache.pinot.core.minion.PinotTaskConfig;
+import org.apache.pinot.spi.annotations.minion.TaskGenerator;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableTaskConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@TaskGenerator
+public class PurgeTaskGenerator implements PinotTaskGenerator {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(PurgeTaskGenerator.class);
+
+    private static final PurgeTaskGenerator INSTANCE = new PurgeTaskGenerator();
+
+    protected ClusterInfoAccessor _clusterInfoAccessor;
+
+    public static Object newInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void init(ClusterInfoAccessor clusterInfoAccessor) {
+        LOGGER.info("I'm in init for the purge");
+        _clusterInfoAccessor = clusterInfoAccessor;
+    }
+
+
+    @Override
+    public String getTaskType() {
+        return MinionConstants.PurgeTask.TASK_TYPE;
+    }
+
+
+
+    @Override
+    public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
+        String taskType = MinionConstants.PurgeTask.TASK_TYPE;
+        List<PinotTaskConfig> pinotTaskConfigs = new ArrayList<>();
+        Set<Segment> runningSegments =
+                TaskGeneratorUtils.getRunningSegments(MinionConstants.PurgeTask.TASK_TYPE, _clusterInfoAccessor);
+        for (TableConfig tableConfig : tableConfigs) {
+            String tableName = tableConfig.getTableName();
```

Comment:
Can we check the table type before getting the task configs? E.g. realtime tables may not have any task configs.

Reply:
Ok I've mooved the check right after for loop declaration and mooved the getRunning task inside it.